src/main/java/org/fenixedu/academic/ui/faces/components/degreeStructure/UICurricularCourse.java

```diff
@@ -207,6 +207,12 @@ private void encodeLoadsAndCredits(CurricularPeriod curricularPeriod) throws IOE
             writer.writeAttribute("align", "right", null);
             writer.append(BundleUtil.getString(Bundle.BOLONHA, "credits.abbreviation")).append(" ");
             writer.append(this.curricularCourse.getEctsCredits(curricularPeriod, executionYear).toString());
+            if (this.curricularCourse.getBaseWeight() != null) {
+                writer.append(" (");
+                writer.append(BundleUtil.getString(Bundle.BOLONHA, "weight")).append(" ");
+                writer.append(this.curricularCourse.getBaseWeight() != null ? this.curricularCourse.getBaseWeight().toString() : "-");
```

Comment:
Because of the enclosing 'if' statement, the this.curricularCourse.getBaseWeight() is never null here